spec/support/with_environment.rb

```diff
@@ -0,0 +1,11 @@
+##
+# temporarily changes the Rails environment
+def with_environment(env_name)
+  prev = Rails.env
+  Rails.env = env_name
+
+  yield
+
+ensure
+  Rails.env = prev
+end
```

Comment:
Would this differ in behavior from the approach we currently use?
```
allow(Rails).to receive(:env).and_return("production".inquiry)
```
We probably want to convert to a string and then inquiry to ensure that existing methods of checking environment wouldn't fail, such as `Rails.env.production?`

Reply:
Using `allow` stubs a specific function for the duration of the subsequent tests. It doesn't feel thorough to me. I don't have any confidence that invisible dependencies are being met.

By setting the .env directly, it changes the environment (and brings along all the rest of the functionality). So setting .env to 'foobar' means that Rails.env.foobar? works as expected. Since at any given time, I'm not certain what underlying structure is depending on the `env`, I preferred to set it at the top level so no discrepancies would pop up.